crates/worker/src/purgatory.rs

```diff
@@ -15,38 +22,58 @@ pub struct Purgatory {
 }
 
 impl Purgatory {
-    pub fn new(state: State) -> Self {
+    pub fn new(state: State, db_client: DynamoDBClient, s3_client: S3Client) -> Self {
         let mut handle = NonNull::dangling();
         let this = Self {
-            inner: Arc::new(Mutex::new(Inner::new(handle, state))),
+            inner: Arc::new(Mutex::new(Inner::new(handle, state, db_client, s3_client))),
         };
 
-        let initialized_handle = tokio::spawn(this.clone().deamon());
-        unsafe {
-            *handle.as_mut() = initialized_handle;
+        {
+            let mut inner = this.inner.try_lock().unwrap();
+            let mut initialized_handle = tokio::spawn(this.clone().deamon());
+            inner.handle = unsafe { NonNull::new_unchecked(&mut initialized_handle as *mut _) };
         }
 
         this
     }
 
     pub async fn purge(&mut self) {
-        self.inner.lock().await.purge()
+        self.inner.lock().await.purge().await;
     }
 
-    pub async fn add_task(&mut self, _: &SqsMessage) {
-        todo!()
+    pub async fn add_record(&mut self, id: Uuid, result: TaskResult) {
+        self.inner.lock().await.add_record(id, result);
     }
 
-    // TODO: args: status, id
-    pub async fn update_task(&mut self) {}
+    async fn deamon(mut self) {
+        const PURGE_INTERVAL: Duration = Duration::from_secs(60);
 
-    async fn deamon(self) {
-        todo!()
+        loop {
+            self.purge().await;
+            sleep(PURGE_INTERVAL).await;
```

Comment:
Use `tokio::time:interval` here instead of sleep

Reply:
Thanks, didn't now about that.

Reply:
Done